docs/proposals/cluster-autoscaling.md

```diff
@@ -0,0 +1,324 @@
+
+```
+                          +--MIG-----------------------------+
+                          |                                  |
+                          | +--------+ +--------+ +--------+ |
+                          | |        | |        | |        | |
+                          | | Node1  | | Node2  | | NodeX  | |
+                          | |        | |        | |        | +<--+
+                          | +--------+ +--------+ +--------+ |   |
+            +---------+   | |cAdvisor| |cAdvisor| |cAdvisor| |   |
+            |  Config |   | +--------+ +--------+ +----+---+ |   |
+            +---------+   |     |          |           |     |   |
+                 |        +-----|----------|-----------|-----+   |
+                 |              |          |           |         |
+            +----v----+         |      +---v----+      |         |
+            |   KAC   |         +----> |Heapster| <----+         |
+            +---------+                +---+----+                |
+                 |                         |                     |
+ Kubernetes      |                         |                     |
+-----------------|-------------------------|---------------------|------
+  Google         |                         |                     |
+                 |                         |                     |
+                 |                  +------v-----+       +-------+------+
+                 |                  |Google Cloud|       |MIG Definition|
+                 |                  | Monitoring |       +-------^------+
+                 |                  +------------+               |
+                 |                         |                     |
+                 |                         |                     |
+                 |                  +------v-----+               |
+                 +----------------->|Google Cloud+---------------+
+                                    | Autoscaler |
+                                    +------------+
+```
+
+
+We will also configure GCE Autoscaler so that it deletes nodes  in a non-random way. There are couple strategies for killing to be considered:
+
+* Node with the fewest number of pods.
+* Node with the fewest number of non-replicated pods.
+* Node with the lowest load.
+* Node with fewest number of guaranteed-QoS pods.
+
+GCA will know about our preferences from the metrics that would go to GCM the similar way as cpu usage (it will kill nodes starting from 
+the one with the lowest metric). As for 1.2 the corresponding API (for selecting which metric to use to choose the victim) 
+may not be ready we will use a predefined metric name and suffix for GCA name.
+We will use MIG shutdown script to support pod graceful termination in case of node deletion.
+
+This will solve the following problems:
+
+* (I1) [GCE specific API] we will have a single tool for all autoscalers (kubectl) with defined configuration
```

Comment:
Maybe add a sentence like "(note that we would still need to adapt KAC to use each cloud provider's autoscaler API)"